utilities/blob_db/blob_db_impl_filesnapshot.cc

```diff
@@ -72,20 +77,26 @@ Status BlobDBImpl::GetLiveFiles(std::vector<std::string>& ret,
   ret.reserve(ret.size() + blob_files_.size());
   for (auto bfile_pair : blob_files_) {
     auto blob_file = bfile_pair.second;
-    ret.emplace_back(blob_file->PathName());
+    // Path should be relative to db_name, but begin with slash.
+    ret.emplace_back(
+        BlobFileName("/" + bdb_options_.blob_dir, blob_file->BlobFileNumber()));
   }
   return Status::OK();
 }
 
 void BlobDBImpl::GetLiveFilesMetaData(std::vector<LiveFileMetaData>* metadata) {
+  // Path should be relative to db_name.
+  assert(bdb_options_.path_relative);
   // Hold a lock in the beginning to avoid updates to base DB during the call
   ReadLock rl(&mutex_);
   db_->GetLiveFilesMetaData(metadata);
   for (auto bfile_pair : blob_files_) {
     auto blob_file = bfile_pair.second;
     LiveFileMetaData filemetadata;
     filemetadata.size = blob_file->GetFileSize();
-    filemetadata.name = blob_file->PathName();
+    // Path should be relative to db_name, but begin with slash.
+    filemetadata.name =
+        BlobFileName("/" + bdb_options_.blob_dir, blob_file->BlobFileNumber());
```

Comment:
I don't like this change. I believe appending a "/" is outside the scope of this function ... may be it should be handled by BlobFileName itself?
Also, we should test whether a trailing "/" is present, before forcing a "/" here.

Reply:
I think it might be better to push the check for handling "/" to be inside util/filename.cc::BlobFileName(), when calling MakeFileName.